README.md

```diff
@@ -301,13 +310,99 @@ var authClient = new OktaAuth(config);
 
 ### Configuration options
 
-These configuration options can be included when instantiating Okta Auth JS (`new OktaAuth(config)`) or in `token.getWithoutPrompt`, `token.getWithPopup`, or `token.getWithRedirect` (unless noted otherwise). If included in both, the value passed in the method takes priority.
+Most configuration options can be included when instantiating Okta Auth JS (`new OktaAuth(config)`) or in `token.getWithoutPrompt`, `token.getWithPopup`, or `token.getWithRedirect` (unless noted otherwise). If included in both, the value passed in the method takes priority.
+
+**Important:** These configuration options can **only** be used when instantiating Okta Auth JS.
+
+#### `storageManager`
+
+The `storageManager` provides access to client storage for specific purposes. `storageManager` configuration is divided into named sections. The default configuration is shown below:
+
+```javascript
+
+var config = {
+  storageManager: {
+    token: {
+      storageTypes: [
+        'localStorage',
+        'sessionStorage',
+        'cookie'
+      ],
+      useMultipleCookies: true // puts each token in its own cookie
+    },
+    cache: {
+      storageTypes: [
+        'localStorage',
+        'sessionStorage',
+        'cookie'
+      ]
+    },
+    transaction: {
+      storageTypes: [
+        'sessionStorage',
+        'localStorage',
+        'cookie'
+      ]
+    }
+  }
+}
+```
+
+**Important:** If neither [localStorage][] nor [sessionStorage][] are available, the default storage provider may fall back to using [cookie][] storage on some clients, . If your site will always be served over a HTTPS connection, you may want to forcibly enable "secure" cookies. This option will prevent cookies from being stored on an HTTP connection.
+
+```javascript
+var config = {
+  cookies: {
+    secure: true
+  }
+}
+```
+
+##### `storageType`
+
+The following values for `storageType` are recognized:
+
+* `memory`: values are stored in a closure and will not survive a page reload
+* `sessionStorage`: will only be available to the current browser tab
+* `localStorage`: available to all browser tabs
+* `cookie`: available to all browser tabs, and server-side code
+
+##### `storageProvider`
```

Comment:
Per #606, looks like users are confused about how the custom storage (storageProvider) is used internally by auth-js. Probably, we can add some explanation for that.